packages/rest-explorer/src/rest-explorer.controller.ts

```diff
@@ -29,11 +30,12 @@ export class ExplorerController {
     @inject(RestBindings.Http.REQUEST) private request: Request,
     @inject(RestBindings.Http.RESPONSE) private response: Response,
   ) {
+    this.serverRootPath = this.getServerRootPath(restConfig);
     this.openApiSpecUrl = this.getOpenApiSpecUrl(restConfig);
   }
 
   indexRedirect() {
-    this.response.redirect(301, this.request.url + '/');
+    this.response.redirect(301, this.serverRootPath + this.request.url + '/');
```

Comment:
I'm not sure if inferring from `openapi.servers[0].url` is a feasible solution. Please note that LoopBack app is not aware how its endpoints are proxies/exposed. The same endpoint can be possibly exposed as many different urls, such as:

- https://a.example.com/api/customer
- https://b.example.com/api/user
- http://internal.example.com:8080/customer

When it comes to redirect via `Location` header, my understanding is that proxy/load balancer is responsible for rewriting `Location` to match its clients.

Reply:
The redirect could be made to depend on the proxy (though that requires config on the proxy).  But the same question would then apply to the `/openapi.json` reference too.

I was concerned about this in my original bug report, but @bajtos had a different perspective.

A brainstorm: I think LB2 dealt with this by having the explorer be the one that serves the `openapi.json`, at least when "talking to itself".  This could be replicated within `rest-explorer` with a little work I think, though it would require the `RestServer` to expose `_serveOpenApiSpec` as a public method.  But I think then the explorer could be made to use exclusively relative URLs, avoiding the path rewriting issue in a more general way?

Reply:
Some references:
- http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect

Reply:
I was hoping that using a relative `Location` url would mean the proxy doesn't _need_ to rewrite it :)

Reply:
I believe that `this.request.url` contains only the path portion of the requested URL _as seen by the LB4 server_, after it has been rewritten by the reverse proxy. The current implementation is correct, it's redirecting from `/explorer` to `/explorer/index.html`.

I am not able to fully understand implications of the change proposed here. More importantly, I don't understand why this particular change is needed.

Can we revert it please?